repondeur/zam_repondeur/views/download.py

```diff
@@ -55,6 +62,47 @@ def download_amendements(context: LectureResource, request: Request) -> Response
         return response
 
 
+@view_config(context=LectureResource, name="export_pdf")
+def export_pdf(context: LectureResource, request: Request) -> Response:
+
+    lecture = context.model(
+        joinedload("articles").joinedload("amendements").joinedload("children")
+    )
+
+    try:
+        nums: List[int] = [int(num) for num in request.params.getall("nums")]
+    except ValueError:
+        raise HTTPBadRequest()
+
+    amendements = [
+        amendement
+        for amendement in (lecture.find_amendement(num) for num in nums)
+        if amendement is not None
+    ]
+
+    with NamedTemporaryFile() as file_:
+
+        tmp_file_path = os.path.abspath(file_.name)
+
+        write_pdf_multiple(
+            lecture=lecture,
+            amendements=amendements,
+            filename=tmp_file_path,
+            request=request,
+        )
+
+        response = FileResponse(tmp_file_path)
+        attach_name = (
+            f"amendement{'s' if len(nums) > 1 else ''}-"
+            f"{','.join(str(num) for num in nums)}-"
+            f"{lecture.chambre}-{lecture.session}-{lecture.num_texte}-"
+            f"{lecture.organe}.pdf"
+        )
+        response.content_type = "application/pdf"
+        response.headers["Content-Disposition"] = f"attachment; filename={attach_name}"
+        return response
+
+
 @view_config(context=AmendementResource, name="download_amendement")
 def download_amendement(context: AmendementResource, request: Request) -> Response:
```

Comment:
@ronnix did you keep that view intentionally?

Reply:
I almost removed it, but I wondered if we would need it, e.g. for printing a single one from my table ?

Happy to remove it if there's no use case...

Reply:
I think I'll remove it. If we need it again, we could even reuse the other view, as printing a single amendement is just a special case of printing multiple ones.